pkg/storage/utils/decomposedfs/tree/tree.go

```diff
@@ -416,6 +416,10 @@ func (t *Tree) ListFolder(ctx context.Context, n *node.Node) ([]*node.Node, erro
 
 // Delete deletes a node in the tree by moving it to the trash
 func (t *Tree) Delete(ctx context.Context, n *node.Node) (err error) {
+	path := filepath.Join(n.ParentPath(), n.Name)
+	// remove entry from cache immediately to avoid inconsistencies
+	_ = t.idCache.Delete(path)
```

Comment:
Little nitpick: Wouldn't it be safer to invalidate the cache after actually having deleted the file? Otherwise the cache could be filled again by another goroutine or instance between here and line 499, no?